### DISCUSSION TITLE
How do I specify the path to the generated fig

### DISCUSSION THREAD

**Discussion by nolanjj** [quarto-dev/quarto-cli#2767] [Q&A]
Help! when I render qmd, The generated fig is located in the figure-gfm folder inside the folder with the same name,
I want the generated fig is located in the  folder with the same name.
What should I do

**Comment by mcanouil** [comment]
If you are using `knitr` as your main engine, then the `fig-path` code cell yaml option (See <https://yihui.org/knitr/options/#plots>) option is what you are looking for.

**Comment by nolanjj** [comment]
Thank you so much！！If I don't use fig-path ，when I render qmd，the fig will save in the "same name_files",
How do I set  to remove the _files suffix

**Reply by mcanouil** [reply]
Far as I know, you can't because the "_files" directory is how and where the HTML page/website finds all the assets (not only figures) it needs to render properly.
So to avoid breaking your document, I would suggest to only change the figures location.

**Reply by mcanouil** [reply]
This is related to #2775

**Reply by nolanjj** [reply]
> Far as I know, you can't because the "_files" directory is how and where the HTML page/website finds all the assets (not only figures) it needs to render properly. So to avoid breaking your document, I would suggest to only change the figures location.

Thank you for your answer。I find a way to solve my problem。after generate md and image
I run 
`rename 's/_files//' *_files`
`ls *.md |while read id;do(cat $id | sed -i 's/_files//' $id );done`

so that I can remove "_files"

Nice!

**Reply by mcanouil** [reply]
I am not sure about this.
It's pretty unsafe and can have unexpected results.

Could you explain in details your use case?

**Reply by nolanjj** [reply]
> I am not sure about this. It's pretty unsafe and can have unexpected results.
> 
> Could you explain in details your use case?

because I want to upload my.md files and images to my Hexo blog
If the name of the folder where the image is located is different from the md file name
image cannot be displayed

So I must keep the name the same
remove "_files" suffix

**Reply by mcanouil** [reply]
Are you using Python/Julia that you cannot simply use the `fig-path` option?

**Reply by nolanjj** [reply]
> Are you using Python/Julia that you cannot simply use the `fig-path` option?

I use the output-dir in _ quarto.yml so that my  .qmd and the output .md are not in the same folder. I have tried that  if Iuse the fig-path option,  finally the image path is not correct.
![image](https://user-images.githubusercontent.com/35267241/194879518-33bfa1d7-5684-4ec5-a4aa-d47fb1bbbbc5.png)

**Reply by mcanouil** [reply]
You need to use the following syntax either in a global `_quarto.yml` or in your Quarto documents header:
```yml
knitr:
  opts_chunk:
    fig.path: "directory/"
```

edit: sorry for the typo `fig.path` for `knitr` chunk option.

**Reply by ejjunju** [reply]
This is not working!

**Reply by cderv** [reply]
Options passed to `opts_chunk` are **knitr** options only and **knitr** uses dot `.` and not dash `-` for option name. 

So try 
````yaml
knitr:
  opts_chunk:
    fig.path: "directory/"
````

Note that this is **knitr** only so your document won't be cross language anymore. Also with more complex project likes book and website, there may be some side effects and features may not correctly support images being moved. 

Quarto has no global `fig-path`  option for a reason: It is really hard to resolve images and paths if we allow configuring this. We choose not too for now. 

Other solution: do this yourself where you need. This means in computation cells do the saving of the graphics where you want them, and then include the images using markdow  syntax where they should be. But not as convenient I acknowledge.

**Reply by mcanouil** [reply]
Apologies for the typo, I am starting to get so used to have dash and no dot in Quarto options^^
(Note, I got it right in #4254 😇)

**Comment by marcdotson** [comment]
What's the equivalent for rendering figures using Python? See #12056.
